fabtests/multinode/include/coll_test.h

```diff
@@ -34,12 +34,14 @@
 
 
 typedef int (*coll_test_setup_t)();
+typedef int (*coll_test_query_t)();
 typedef int (*coll_test_run_t)();
 typedef void (*coll_test_teardown_t)();
 
 struct coll_test {
         char *name;
         coll_test_setup_t setup;
+        coll_test_query_t query;
```

Comment:
I like where these changes are going, but I'd go further and update how the test is structured.  I'd start by changing struct coll_test to this:
```
struct coll_test {
    char *name;
    int (*test_setup)(void);
    int (*test_run)(enum fi_collective_op coll_op, enum fi_op op, enum fi_datatype datatype);
    int (*test_teardown)(void);
    enum fi_collective_op coll_op;
    enum fi_op op;
    enum fi_datatype datatype;
};
```
There's no need for the typedefs, we can define the callbacks directly in the structure, which makes things easier to read.  We also pass in the coll_op, op, datatype into the test_run handlers, which will make them more generic.  This eliminates the query callback needing information about the internal implementation of those calls.  For example, as proposed sum_all_reduce_test_query() must just know how sum_all_reduce_test_run() is implemented.

Instead, just call test_query() with the information from coll_test.  That same information is then passed to allreduce_test_run().

Because sum_all_reduce_test_run() verifies the result against expected data, for now I'd have the function assert that the op and datatype are as expected.  Later we can separate and expand the calculation for expected data to account for other ops and datatypes

Reply:
Done